sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -167,18 +168,38 @@ class MapboxMap internal constructor(
     onStyleLoaded: Style.OnStyleLoaded? = null
   ) {
     this.style = style
-    styleExtension.images.forEach {
-      it.bindTo(style)
-    }
     styleExtension.sources.forEach {
+      if (it is GeoJsonSource) {
+        if (it.geoJsonParsed) {
+          it.bindTo(style)
+          styleExtension.resourceCount--
+        } else {
+          it.onGeoJsonParsedListener = { geoJson ->
+            geoJson.bindTo(style)
+            styleExtension.resourceCount--
+            if (styleExtension.resourceCount == 0) {
+              onStyleLoaded?.onStyleLoaded(style)
+            }
+          }
+        }
```

Comment:
as commented before, I think we can always use this logic branch.

Reply:
I think no. This function is responsible for decreasing the counter and actually does not have any information if geojson was already loaded until this point. 
We could add `onGeoJsonParsedListener` but it was already triggered beforehand.

Reply:
if we make the `geoJsonParsed` private like proposed in  onhttps://github.com/mapbox/mapbox-maps-android/pull/327/files#r634347933 , we could always use the `onGeoJsonParsedListener` and decrease the counter inside the `onGeoJsonParsedListener`.